velox/core/SimpleFunctionMetadata.h

```diff
@@ -278,13 +278,24 @@ class UDFHolder final
   template <typename T>
   using optional_exec_arg_type = std::optional<exec_arg_type<T>>;
 
+  DECLARE_CONDITIONAL_TYPE_NAME(
```

Comment:
Chatted offline.  For non-complex times in_type and null_free_in_type will be the same (at this point only Array, Map, Row, and Variadic have different types here).  Doing this allows us to avoid having to specify the identical type mapping for those types.

For resolvers outside Velox that don't have a null_free_in_type for their complex types, they'll see failures at compilation time since their in_types won't support the right interface.